src/main/java/org/kitteh/irc/client/library/util/StringUtil.java

```diff
@@ -120,4 +124,37 @@ public static String makeRainbow(@Nonnull String message, @Nonnull Format[] colo
         }
         return builder.toString();
     }
+
+    /**
+     * Converts a given String to lowercase per spec.
+     *
+     * @param cliented the client provider
+     * @param input string to be lowercased
+     * @return lowercased string
+     * @throws IllegalArgumentException if client provider is null
+     * @throws IllegalArgumentException if input is null
```

Comment:
Maybe "if either parameter is null" ?

Also, thoughts on nullchecking the result of `#getClient()`? KICL doesn't do it but other things might